### DISCUSSION TITLE
what tools are you useing and if so are they public or going to

### DISCUSSION THREAD

**Discussion by retexcraft** [phst-randomizer/ph-randomizer#233] [Q&A] [answered]
i would like to rom hack this myself and the tools that used to exist no longer are avalibale and so was wander if you guys had some also if i cannot have them i would like to join the project (i mean i would ether way)

**Comment by mike8699** [comment, marked as answer]
Hi! As far as custom tooling goes, we're using [a custom fork of `zed`](https://github.com/phst-randomizer/zed), which is a Python library for working with the various file formats in the NDS Zeldas. For making changes to the game code itself (i.e. "ASM hacking"), we use ARMIPS; [this file](https://github.com/phst-randomizer/ph-randomizer/blob/main/base/code/main.asm) is the entrypoint for that. Both are completely public and open-source.

**Reply by NQNStudios** [reply]
Hey, the link to the ASM entry point file isn't valid anymore. It points to base/src/main.asm, would the current location of that file be base/code/main.asm?

**Reply by mike8699** [reply]
Yes, that's correct.
